packages/strapi/lib/services/metrics/__tests__/rate-limiter.test.js

```diff
@@ -0,0 +1,48 @@
+const wrapWithRateLimiter = require('../rate-limiter');
+
+describe('Telemetry daily RateLimiter', () => {
+  test('Passes event and payload to sender', async () => {
+    const sender = jest.fn(() => Promise.resolve(true));
+
+    const send = wrapWithRateLimiter(sender, { limitedEvents: ['testEvent'] });
+
+    const payload = { key: 'value' };
+    await send('notRestricted', payload);
+
+    expect(sender).toHaveBeenCalledWith('notRestricted', payload);
+  });
+
+  test('Calls sender if event is not restricted', async () => {
+    const sender = jest.fn(() => Promise.resolve(true));
+
+    const send = wrapWithRateLimiter(sender, { limitedEvents: ['testEvent'] });
+
+    await send('notRestricted');
+
+    expect(sender).toHaveBeenCalledWith('notRestricted', undefined);
+  });
+
+  test('Calls the sender as many times as request when events is not restricted', async () => {
```

Comment:
Calls the sender as many times as request**ed** when events **are** not restricted